packages/ra-core/src/controller/ListController.tsx

```diff
@@ -244,52 +233,26 @@ export class UnconnectedListController extends Component<
         return true;
     }
 
-    /**
-     * Check if user has already set custom sort, page, or filters for this list
-     *
-     * User params come from the Redux store as the params props. By default,
-     * this object is:
-     *
-     * { filter: {}, order: null, page: 1, perPage: null, sort: null }
-     *
-     * To check if the user has custom params, we must compare the params
-     * to these initial values.
-     *
-     * @param {object} params
-     */
-    hasCustomParams(params: ListParams) {
-        return (
-            params &&
-            params.filter &&
-            (Object.keys(params.filter).length > 0 ||
-                params.order != null ||
-                params.page !== 1 ||
-                params.perPage != null ||
-                params.sort != null)
-        );
-    }
-
     /**
      * Merge list params from 4 different sources:
      *   - the query string
      *   - the params stored in the state (from previous navigation)
-     *   - the filter defaultValues
      *   - the props passed to the List component
      */
-    getQuery() {
+    getQuery(props = this.props) {
         const query: Partial<ListParams> =
-            Object.keys(this.props.query).length > 0
-                ? this.props.query
-                : this.hasCustomParams(this.props.params)
-                ? { ...this.props.params }
-                : { filter: this.props.filterDefaultValues || {} };
+            Object.keys(props.query).length > 0
+                ? props.query
+                : hasCustomParams(props.params)
+                ? props.params
```

Comment:
It was cloned for a reason (I think because parmas are mutated somewhere), you shouldn't change that.

Reply:
Fixed